.gitlab-ci.yml

```diff
@@ -133,102 +157,78 @@ build:ubuntu-14.04:
     CXX: g++-6
 
 package:ubuntu-14.04:
+  extends: .debpkg_template
   image: ubuntu:trusty
   dependencies:
     - build:ubuntu-14.04
   variables:
     LEVELDB_PKG: libleveldb1
-  <<: *debpkg_template
 
 deploy:ubuntu-14.04:
+  extends: .debpkg_install
   image: ubuntu:trusty
   dependencies:
     - package:ubuntu-14.04
   variables:
     LEVELDB_PKG: libleveldb1
-  <<: *debpkg_install
 
 # Xenial
 
 build:ubuntu-16.04:
-  <<: *build_definition
+  extends: .build_template
   image: ubuntu:xenial
   before_script:
     - apt-get update -y
     - apt-get -y install build-essential libirrlicht-dev cmake libbz2-dev libpng-dev libjpeg-dev libxxf86vm-dev libgl1-mesa-dev libsqlite3-dev libogg-dev libvorbis-dev libopenal-dev libcurl4-gnutls-dev libfreetype6-dev zlib1g-dev libgmp-dev libjsoncpp-dev
 
 package:ubuntu-16.04:
+  extends: .debpkg_template
   image: ubuntu:xenial
   dependencies:
     - build:ubuntu-16.04
   variables:
     LEVELDB_PKG: libleveldb1v5
-  <<: *debpkg_template
 
 deploy:ubuntu-16.04:
+  extends: .debpkg_install
   image: ubuntu:xenial
   dependencies:
     - package:ubuntu-16.04
   variables:
     LEVELDB_PKG: libleveldb1v5
-  <<: *debpkg_install
 
 # Yakkety
 
 #build:ubuntu-16.10:
-#  <<: *build_definition
+#  extends: .build_template
 #  image: ubuntu:yakkety
 #  before_script:
 #    - apt-get update -y
 #    - apt-get -y install build-essential libirrlicht-dev cmake libbz2-dev libpng-dev libjpeg-dev libxxf86vm-dev libgl1-mesa-dev libsqlite3-dev libogg-dev libvorbis-dev libopenal-dev libcurl4-gnutls-dev libfreetype6-dev zlib1g-dev libgmp-dev libjsoncpp-dev
 
 #package:ubuntu-16.10:
+#  extends: .debpkg_template
 #  image: ubuntu:yakkety
 #  dependencies:
 #    - build:ubuntu-16.10
 #  variables:
 #    LEVELDB_PKG: libleveldb1v5
-#  <<: *debpkg_template
 
 #deploy:ubuntu-16.10:
+#  extends: .debpkg_install
 #  image: ubuntu:yakkety
 #  dependencies:
 #    - package:ubuntu-16.10
 #  variables:
 #    LEVELDB_PKG: libleveldb1v5
```

Comment:
this is all also commented out, just remove it

Reply:
yep good idea. We should rebuild up to date ubuntu images at a point